ci: pull request template (#517)
opentargets/gentropy: .github/pull_request_template.md

## ✨ Context

> <!--- What's the context for the changes? If the changes are related to a specific issue, please [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to it:_

Reviewer: ```suggestion
<!---
Congratulations! You've made it this far!

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your contribution.

What's the context for the changes? If the changes are related to a specific issue, please [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to it:_
```

- This PR fixes or closes issue: fixes #
- This PR is related to issue:

-->

## 🛠 What does this PR implement

<!--- _Detailed description of the changes introduced, Give examples of the changes you've made in this pull request, include an itemized list if you can and
add diagrams or images if necessary. It'll help the reviewer_ -->

## 🙈 Missing

<!--- If there are things that are requested in the task and were not implemented, list them here -->

> ## 🚦 Checks

Reviewer: ```suggestion
## 🚦 Before submitting
```

> - [ ] covers one single feature (one change at a time)

Reviewer: ```suggestion
- [ ] Do these changes cover one single feature (one change at a time)?
```

> - [ ] documentation added/updated

Reviewer: ```suggestion
- [ ] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
-  [ ] Did you make sure to update the documentation with your changes?
```

> - [ ] there is no commented out code in this PR

Reviewer: ```suggestion
- [ ] Did you make sure there is no commented out code in this PR?
```

> - [ ] follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) in PR title and commits.

Reviewer: ```suggestion
- [ ] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
```

> - [ ] is up-to-date with `dev` branch

Reviewer: ```suggestion
- [ ] Did you make sure the branch is up-to-date with the `dev` branch?
```

> - [ ] tests have been added to verify that the new code works

Reviewer: ```suggestion
- [ ] Did you write any new necessary tests?
```

> - [ ] passes local tests (`make test`)

Reviewer: ```suggestion
- [ ] Did you make sure the changes pass local tests (`make test`)?
```

> - [ ] passes pre-commits (e.g `poetry run pre-commit run --all-files`)

Reviewer: ```suggestion
- [ ] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?
```
